imaging/imageRendering.ts

```diff
@@ -935,33 +948,40 @@ let getSeriesData = function (
   data.thickness = series.instances[series.imageIds[0]].metadata[
     "x00180050"
   ] as number;
-
   let spacing = series.instances[series.imageIds[0]].metadata[
     "x00280030"
   ] as number[];
   data.spacing_x = spacing ? spacing[0] : 1;
   data.spacing_y = spacing ? spacing[1] : 1;
-
   // window center and window width
-  data.wc =
-    defaultProps && defaultProps.wc !== undefined
-      ? defaultProps["wc"]
-      : (series.instances[series.imageIds[0]].metadata["x00281050"] as number);
-
-  data.ww =
-    defaultProps && defaultProps.ww !== undefined
-      ? defaultProps["ww"]
-      : (series.instances[series.imageIds[0]].metadata["x00281051"] as number);
-
-  // default values for reset
-  data.defaultWW =
-    defaultProps && has(defaultProps, "defaultWW")
-      ? defaultProps["defaultWW"]
-      : data.ww;
-  data.defaultWC =
-    defaultProps && has(defaultProps, "defaultWC")
-      ? defaultProps["defaultWC"]
-      : data.wc;
+  data.viewport = {
+    voi: {
+      windowCenter:
+        defaultProps && has(defaultProps, "wc")
+          ? defaultProps.wc
+          : (series.instances[series.imageIds[0]].metadata[
+              "x00281050"
+            ] as number),
+      windowWidth:
+        defaultProps && has(defaultProps, "ww")
+          ? defaultProps.ww
+          : (series.instances[series.imageIds[0]].metadata[
+              "x00281051"
+            ] as number)
+    }
+  };
+  data.default = {
+    voi: {
+      windowCenter:
+        defaultProps && has(defaultProps, "defaultWC")
+          ? defaultProps.defaultWC
+          : data.viewport!.voi!.windowCenter,
+      windowWidth:
+        defaultProps && has(defaultProps, "defaultWW")
+          ? defaultProps.defaultWW
+          : data.viewport!.voi!.windowWidth
+    }
+  };
```

Comment:
il fatto di aver spostato gli oggetti sotto "voi" lo rende retrocompatibile con applicazioni che usano versione vecchie di larvitar? E' usato anche da altre parti? (credo di no ma chiedo per conferma)

Reply:
nono, è solo la costruzione di un oggetto interno a una funzione non esposta, l'ho fatto solo per non scrivere un tipo in più visto che l'oggetto è molto simile a uno esistente